runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -427,10 +428,22 @@ public void reconfigureSequencerServers(Layout originalLayout, Layout newLayout,
                         || !originalLayout.getPrimarySequencer()
                         .equals(newLayout.getPrimarySequencer())) {
 
+                    // The tails query and the sequencer recovery must be performed on the same
+                    // epoch. If they are not, we lose the atomicity of bootstrapping the sequencer
+                    // which can lead to data operations across epochs leaking into the
+                    // Address space causing inconsistencies and data loss.
+
                     //TODO(Maithem) why isn't this getting the tails
```

Comment:
Now that you're cleaning this code, can you take care of this pending TODO, from what I see addressView is getting It from Utils but through the LayoutHelper... if we do it directly from Utils would we guarantee it throws a WrongEpochException if not on the same epoch as runtime? (assuming it is on newLayout, am I right?)

Reply:
+1 please try to do the clean up if you can

Reply:
Done.